drivers/include/hcsr04.h

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (C) 2021 AUTH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_hcsr04 HC-SR04
+ * @ingroup     drivers_sensor
+ * @ingroup     drivers_saul
+ * @brief       This is a sensor which uses ultrasonic waves in order to measure the distance
+ *
+ * @{
+ *
+ * @file
+ * @brief       This is a sensor which uses ultrasonic waves in order to measure the distance
+ *
+ * @author      Evripidis Chondromatidis <eurichon1996@gmail.com>
+ */
+
+#ifndef HCSR04_H
+#define HCSR04_H
+
+#include <stdint.h>
+#include "hcsr04_constants.h"
+#include "periph/gpio.h"
+#include "mutex.h"
+
+
+/* Add header includes here */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Declare the API of the driver */
+
+/**
+ * @brief   Device initialization parameters for hcsr04 sensor
+ */
+typedef struct {
+    uint16_t temperature;
+    gpio_t trigger_pin;
+    gpio_t echo_pin;
+} hcsr04_params_t;
+
+/**
+ * @brief   Device descriptor of a HC-SR04 sensor
+ */
+typedef struct {
+    mutex_t lock;
+    uint8_t was_triggered;
+    uint16_t distance;
+    uint16_t sound_speed;
+    uint32_t pre_trig_t_us;
+    uint32_t pre_meas_t_us;
+    hcsr04_params_t params;
+} hcsr04_t;
+
+/**
+ * @brief   Initialize an HC-SR04 sensor
+ *
+ * @param[inout] dev        Device descriptor of the driver
+ * @param[in]    params     Initialization parameters
+ *
+ * @return                  EINVAL of invalid parameters
```

Comment:
```suggestion
 * @retval                  EINVAL of invalid parameters
```

Same for other `@return`s with multiple values (>1) here and in other functions